Fix issues with draft submission and history display

paulfalgout opened the pull request:
Shortcut Story ID: [sc-40419] [sc-40447]

Includes a bit of clean up, but largely adds an existing id to draft or submissions if there's a latestDraft.

paulfalgout committed:
Rename getSubmission to indicate it returns the first

paulfalgout committed:
Filter drafts from form response history

shadowhand commented on src/js/entities-service/entities/form-responses.js:
```diff
@@ -58,15 +53,16 @@ const Collection = BaseCollection.extend({
     return this.find({ status: FORM_RESPONSE_STATUS.SUBMITTED });
   },
+  filterSubmissions() {
+    const submissions = this.filter({ status: FORM_RESPONSE_STATUS.SUBMITTED });
+
+    const clone = this.clone();
+    clone.reset(submissions);
+    return clone;
```
```suggestion
    const clone = this.clone();
    const submissions = this.filter({ status: FORM_RESPONSE_STATUS.SUBMITTED });

    clone.reset(submissions);

    return clone;
```

paulfalgout committed:
Remove unused code

Latest draft filtering by editor is happening on the endpoint due to patient forms not having a list of responses.

paulfalgout committed:
Patch to drafts for updating drafts or submissions
